src/media/js/elements/select.js

```diff
@@ -0,0 +1,260 @@
+      internal <select>, interfaces like HTMLSelectElement.selectedIndex and
+      HTMLSelectElement.value will automatically work.
+    - <mkt-select> visibility is toggled on click (.mkt-select--visible).
+    - <mkt-option> elements have their padding-left programmatically set to
+      all align with the <mkt-selected-text>.
+    - <mkt-option> elements keep their data-mkt-option--index up-to-date so
+      that colors can be set appropriate on each while ignoring the selected
+      option.
+    - To change <mkt-select>'s value, do mktSelect.value = 'newValue';
+*/
+define('elements/select',
+    ['core/log', 'core/z', 'jquery', 'underscore'],
+    function(log, z, $, _) {
+    'use strict';
+    var logger = log('elements/select');
+
+    var VISIBLE = 'mkt-select--visible';
+
+    var MktOptionElement = document.registerElement('mkt-option', {
+        prototype: Object.create(HTMLOptionElement.prototype, {
+            option: {
+                // Actual <option> element to proxy to, steal its interface.
+                // Value set in <mkt-select>'s createdCallback.
+                value: null,
+                writable: true
+            },
+            update: {
+                value: function() {
+                    // Update attrs on self, notably [selected].
+                    this.removeAttribute('selected');
+                    copyAttrs(this, this.option);
+                }
+            }
+        })
+    });
+
+    var MktOptGroupElement = document.registerElement('mkt-optgroup', {
+        prototype: Object.create(HTMLOptGroupElement.prototype, {
+            optGroup: {
+                // Actual <optgroup> element to proxy to, steal its interface.
+                // Value set in <mkt-select>'s createdCallback.
+                value: null,
+                writable: true
+            }
+        })
+    });
+
+    var MktSelectElement = document.registerElement('mkt-select', {
+        prototype: Object.create(HTMLSelectElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    // Create a hidden <select> element.
+                    var select = document.createElement('select');
+                    select.style.display = 'none';
+                    copyAttrs(select, root);
+
+                    // Create <mkt-option>/<mkt-optgroup> using <option>s.
+                    var mktOptGroup = document.createElement('mkt-optgroup');
+                    var optGroup = root.querySelector('optgroup');
+                    var options = root.querySelectorAll('option');
+                    for (var i = 0; i < options.length; i++) {
+                        var option = options[i];
+
+                        var mktOption = document.createElement('mkt-option');
+                        mktOption.option = option;
+                        mktOption.innerHTML = option.innerHTML;
+                        mktOption.update();
```

Comment:
If you define a setter on `mktOption.option` then you could put this all in `MktOptionElement`.

```js
var mktOption = document.createElement('mkt-option');
mktOption.option = option;
mktOptGroup.appendChild(mktOption);